docs/error-codes.md

```diff
@@ -1511,3 +1511,29 @@ This is technically possible if a custom assembly defines `DynamicDependencyAttr
   [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
   object TestProperty { get; set; }
   ```
+
+### 'IL2100': XML may not contain wildcard for assembly fullname
+
+- A wildcard "fullname" for an assembly in XML is only valid for link attribute XML on the command-line, not for descriptor or substitution XML or for embedded attribute XML. Specify a specific assembly name instead.
+
+  ```XML
+  <!-- IL2100: XML may not contain wildcard for assembly fullname -->
+  <linker>
+    <assembly fullname="*">
+      <type fullname="MyType" />
+    </assembly>
+  </linker>
+  ```
+
+### 'IL2101': Embedded XML in assembly 'assembly' may not modify other assembly 'assembly'.
+
+- Embedded attribute or substitution XML may only contain elements that apply to the containing assembly. It is invalid to use these embedded XML files to modify other assemblies.
+
+  ```XML
+  <!-- IL2101: Embedded XML in assembly 'ContainingAssembly' may not modify other assembly 'OtherAssembly' -->
+  <linker>
+    <assembly fullname="OtherAssembly">
```

Comment:
fullname is now redundant for embedded XML format, should we just remove/ignore it instead?

Reply:
I think it's good to warn when it's used incorrectly in the embedded XML, otherwise people might try to embed existing XML and wonder why it doesn't work.

Since the assembly is redundant for embedded XML, I made it so that you can leave out the `assembly` entirely:
```xml
<linker>
  <type fullname="..." />
</linker>
```

I kept support for `<assembly fullname="ThisAssembly>"` since I think it will be easier to reuse existing XML this way, but I could force the embedded XML not to name the assembly if you prefer.

Reply:
Right, I think this is more wording issue than anything else. Your "It is invalid to use these embedded XML files to modify other assemblies." seems to suggest this will be error where in reality this is a warning. Maybe we could go with even simple mode and merge 2100 and 2101 under the simple warning that warns that `fullname` value is ignored for embedded XML files (just suggestion)

Reply:
I reworded the warnings - hopefully this is better. I'd like to keep them separate because the wildcard can produce a warning for non-embedded XML too.

Reply:
> fullname is now redundant for embedded XML format, should we just remove/ignore it instead?

My name is Scarlett Kelley

Reply:
> I think it's good to warn when it's used incorrectly in the embedded XML, otherwise people might try to embed existing XML and wonder why it doesn't work.
> 
> Since the assembly is redundant for embedded XML, I made it so that you can leave out the `assembly` entirely:
> ```xml
> <linker>
>   <type fullname="..." />
> </linker>
> ```
> 
> I kept support for `<assembly fullname="ThisAssembly>"` since I think it will be easier to reuse existing XML this way, but I could force the embedded XML not to name the assembly if you prefer.

My full name is Scarlett Kelley